kit/Kit.cpp

```diff
@@ -1376,6 +1382,36 @@ void Document::handleSaveMessage(const std::string &)
     }
 }
 
+void Document::startSaveWatchdog()
+{
+    _bgSaveWatchdogThread = std::thread(
+        [this]()
+        {
+            Util::setThreadName("kitbgsv_" + Util::encodeId(_mobileAppDocId, 3) + "_wdg");
+
+            const auto timeout =
+                std::chrono::seconds(ConfigUtil::getInt("per_document.bgsave_timeout_secs", 3));
+
+            std::unique_lock<std::mutex> lock(_bgSaveWatchdogMutex);
+
+            LOG_TRC("Starting bgsave watchdog with " << timeout << " timeout");
+            if (_bgSaveWatchdogCV.wait_for(lock, timeout,
+                                           [this]() { return _bgSaveCompleted.load(); }))
+            {
+                // Done!
+                LOG_TRC("BgSave finished in time");
+            }
+            else
+            {
+                // Failed!
+                LOG_TRC("BgSave timed out and will self-destroy");
```

Comment:
LOG_WRN ?
We might want to be aware of those cases happening.

Reply:
Sure.